acq_stat_reports/acq_stat_reports.py

```diff
@@ -134,184 +111,209 @@ def make_acq_plots(acqs, tstart=0, tstop=None, outdir=None, close_figures=False)
+        [np.linspace(5.3, 8.3, 16), [8.5, 8.7, 8.9, 9.2, 9.5, 9.8, 10.2, 11, 12]]
     )
-    plt.errorbar(
-        x, fracs, yerr=[err_low, err_high], color="black", marker=".", linestyle="None"
-    )
-    (x, fracs, err_low, err_high) = frac_points(
-        acqs,
-        mag_bin,
-        binstart=np.arange(10 - (mag_bin / 2), 11 + (mag_bin / 2), mag_bin),
-    )
-    plt.errorbar(
-        x, fracs, yerr=[err_low, err_high], color="red", marker=".", linestyle="None"
-    )
-    plt.ylim(-0.05, 1.05)
-    plt.xlim(10, 11)
-    plt.xlabel("Star magnitude (mag)")
-    plt.ylabel("Fraction Acquired")
-    plt.title("Acquisition Success vs Expected Mag")
-    plt.subplots_adjust(top=0.85, bottom=0.17, right=0.97)
-    if outdir:
-        plt.savefig(outdir / "zoom_mag_pointhist.png")
-    if close_figures:
-        plt.close(h)
-
-    h = plt.figure(figsize=figsize)
-    plt.hist(
-        range_acqs["mag"],
-        bins=np.arange(5.5 - (mag_bin / 2), 12 + (mag_bin / 2), mag_bin),
-        histtype="step",
-        color="black",
-        density=True,
-    )
-    plt.hist(
-        acqs["mag"],
-        bins=np.arange(5.5 - (mag_bin / 2), 12 + (mag_bin / 2), mag_bin),
-        histtype="step",
-        color="red",
-        density=True,
-    )
-    plt.xlabel("Star magnitude (mag)")
-    plt.ylabel("Fraction of All Acq Stars")
-    plt.title("Expected Magnitudes of Acquisition Stars")
-    plt.ylim(0, 1.0)
-    plt.subplots_adjust(top=0.85, bottom=0.17, right=0.97)
-    if outdir:
-        plt.savefig(outdir / "exp_mag_histogram.png")
-
-    plt.figure(figsize=figsize)
-    ok = range_acqs["acqid"] == 1
+
+    datasets = {
+        "all_acq": range_acqs,
+        "two_year_acqs": two_year_acqs,
+        "binned_p_acq": utils.BinnedData(
+            data=range_acqs,
+            bins={"p_acq_model": np.linspace(0, 1, 10)},
+        ),
+        "binned_t_ccd": utils.BinnedData(
+            data=range_acqs,
+            bins={"t_ccd": t_ccd_bins},
+        ),
+        "variable_binned_mag": utils.BinnedData(
+            data=range_acqs,
+            bins={"mag": mag_bins},
+        ),
+        "binned_mag": utils.BinnedData(
+            data=range_acqs,
+            bins={"mag": np.arange(5.5 - (0.2 / 2), 12 + (0.2 / 2), 0.2)},
+        ),
+        "binned_time": utils.BinnedData(
+            data=two_year_acqs,
+            bins={"tstart": np.arange(tstop, tstop - 2 * 365 * 86400, -86400 * 30)},
+        ),
+    }
+
+    functions = {
+        "acq_stars_plot": acq_stars_plot,
+        "mag_scatter_plot": mag_scatter_plot,
+        "utils.binned_data_fraction_plot": utils.binned_data_fraction_plot,
+        "utils.binned_data_plot": utils.binned_data_plot,
+        "utils.binned_data_probability_plot": utils.binned_data_probability_plot,
+        "fail_rate_plot": fail_rate_plot,
+    }
+
+    plot_params = {
+        "acq_stars": {
+            "data": "two_year_acqs",
+            "class": "acq_stars_plot",
```

Comment:
Since these are no longer classes, maybe it is worth the time to relabel them as funcs or whatever.